File: 003_Digit_Counts.java
public class Solution {
    /*
     * @param : An integer
     * @param : An integer
     * @return: An integer denote the count of digit k in 1..n
     */
      public int digitCounts(int k, int n) {
        int count = 0;
        char kChar = (char)(k + '0');
        for (int i = k; i <= n; i++) {
            char[] iChars = Integer.toString(i).toCharArray();
            for (char iChar : iChars) {
                if (kChar == iChar) count++;
            }
        }

        return count;
    }
};

File: 003_Digit_Counts.cpp
class Solution {
public:
    /*
     * @param : An integer
     * @param : An integer
     * @return: An integer denote the count of digit k in 1..n
     */
 int digitCounts(int k, int n) {
    // write your code here
    int result = 0;
    for(int i = 0;i<=n;i++){
        int s = i;
        if(s == 0 && k == 0){
            result ++;
        }
        while(s){
            if(s%10 == k){
                result ++;
            }
            s /= 10;
        }

    }
    return result;
}
 
};
